packages/@ourworldindata/grapher/src/lineLegend/LineLegendFilterAlgorithms.ts

```diff
@@ -0,0 +1,140 @@
+import { maxBy, partition } from "@ourworldindata/utils"
+import {
+    computeCandidateScores,
+    LineLegendFilterAlgorithmContext,
+    pickAsManyAsPossibleWithRetry,
+    pickCandidateWithMaxDistanceToReferenceCandidate,
+    pickCandidateWithRetry,
+    pickOutermostCanidate,
+} from "./LineLegendHelpers"
+import { PlacedSeries } from "./LineLegendTypes"
+
+/**
+ * Keep a subset of series that fit within the available height, prioritizing by
+ * importance. Focused series have priority, even if they're less important.
+ *
+ * Note that more important (but longer) series names might be skipped if they don't fit.
+ */
+export function findImportantSeriesThatFitIntoTheAvailableSpace(
+    seriesSortedByImportance: PlacedSeries[],
+    availableHeight: number
+) {
+    let context: LineLegendFilterAlgorithmContext = {
+        candidates: new Set(seriesSortedByImportance),
+        availableHeight,
+        sortedKeepSeries: [],
+        keepSeriesHeight: 0,
+    }
+
+    const [focusedCandidates, nonFocusedCandidates] = partition(
+        seriesSortedByImportance,
+        (series) => series.focus?.active
+    )
+
+    const importanceScore = new Map(
+        seriesSortedByImportance.map((series, index) => [
+            series.seriesName,
+            -index, // higher index means lower importance
+        ])
+    )
+
+    const pop = (candidates: PlacedSeries[]) =>
+        maxBy(candidates, (c) => importanceScore.get(c.seriesName))
```

Comment:
I called it `pop...` because I wanted to communicate that the element is removed from the array after it's been checked, but I can see how that's confusing in a different way. I'll rename to `get...` :)